Hw4

AbylaiNur opened the pull request:
После проверки hw3 я сразу добавлю изменения в hw4 и сделаю push

AbylaiNur committed:
hw4

AbylaiNur commented:
В общем изменений после проверки hw3 не было. Поэтому нового пуша с изменениями нету.

AirEM commented on migrations/chat.sql:
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE chat (
+    id BIGINT PRIMARY KEY,
+    bot_state VARCHAR(255),
```
что за поле ? давай оставим коммент в БД

AirEM commented on migrations/chat.sql:
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE chat (
+    id BIGINT PRIMARY KEY,
+    bot_state VARCHAR(255),
+    bot_last_active_command VARCHAR(255)
```
что за поле ? давай оставим коммент в БД

AirEM commented on migrations/link.sql:
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE link (
+    id BIGINT PRIMARY KEY,
+    url VARCHAR(255) not null,
+    chat_id INT not null,
```
какой вид связи получается ? Точно ли он должен быть таким ?

AirEM commented on migrations/link.sql:
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE link (
+    id BIGINT PRIMARY KEY,
+    url VARCHAR(255) not null,
```
Может уникальность добавим ?

AirEM commented on scrapper/src/test/java/ExampleTest.java:
```diff
@@ -0,0 +1,11 @@
+
+    @Test
+    void containerIsRunningTest() {
+        assertTrue(POSTGRE_SQL_CONTAINER.isRunning());
```
ну как бы да
но хотелось бы протестировать, что миграции накатились например

AirEM commented on docker-compose.yml:
```diff
@@ -0,0 +1,35 @@
+services:
```
предлагаю migrations и docker-compose вынести в модульностью, где они используются

AbylaiNur committed:
Изменения:

 - перенес docker-compose.yml в scrapper

 - добавил тест для проверки создании таблиц миграции

 - добавил chat_link.sql

 - добавил column коментарии в chat.sql, link.sql и chat_link.sql

AirEM commented on scrapper/migrations/chat.sql:
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE chat (
+    id                      BIGINT          PRIMARY KEY,
+    bot_state               VARCHAR(255),
+    bot_last_active_command VARCHAR(255)
```
А зачем это в БД хранить ?

AbylaiNur replied:
Думал если пользователь напишет в час после 2-3 дней в чат, то бот будет знать какую команду обработать. К примеру написал /track, потом через 3 часа написал ссылку и бот добавит ссылку. Или лучше хранить состояние в бэке и не париться по этому поводу или оставить состояние в дб?

AbylaiNur replied:
Идея в том что если сервер упадет, то можно взять последнее состояние бота в бд

AirEM replied:
Состояние на бэке точно не стоит хранить. Это как мнимым ненадежно. 
Ну окей, пусть будет. Я просто сразу не Понял, зачем это поле

AbylaiNur replied:
А ок, тогда пофиксаю это

AbylaiNur committed:
changes:

 - removed bot state and bot last active command from migraions

AbylaiNur committed:
fix
